src/backend/libc/io/syscalls.rs

```diff
@@ -531,3 +533,98 @@ pub unsafe fn vmsplice(
     ))
     .map(|spliced| spliced as usize)
 }
+
+#[cfg(any(target_os = "solaris", target_os = "illumos",))]
+pub(crate) fn port_create() -> io::Result<OwnedFd> {
+    unsafe { ret_owned_fd(c::port_create()) }
+}
+
+#[allow(unused_unsafe)]
+#[cfg(any(target_os = "solaris", target_os = "illumos",))]
+pub(crate) unsafe fn port_associate(
+    port: BorrowedFd<'_>,
+    source: c::c_int,
+    object: c::uintptr_t,
+    events: c::c_int,
+    user: *mut c::c_void,
+) -> io::Result<()> {
+    unsafe {
+        ret(c::port_associate(
+            borrowed_fd(port),
+            source,
+            object,
+            events,
+            user,
+        ))
+    }
+}
+
+#[allow(unused_unsafe)]
+#[cfg(any(target_os = "solaris", target_os = "illumos"))]
+pub(crate) unsafe fn port_dissociate(
+    port: BorrowedFd<'_>,
+    source: c::c_int,
+    object: c::uintptr_t,
+) -> io::Result<()> {
+    unsafe { ret(c::port_dissociate(borrowed_fd(port), source, object)) }
+}
+
+#[cfg(any(target_os = "solaris", target_os = "illumos"))]
+pub(crate) fn port_get(
+    port: BorrowedFd<'_>,
+    timeout: Option<&mut c::timespec>,
+) -> io::Result<Option<Event>> {
+    let mut event = MaybeUninit::<c::port_event>::uninit();
+    let timeout = timeout.map_or(core::ptr::null_mut(), |t| t as *mut _);
+    let result = unsafe { ret(c::port_get(borrowed_fd(port), event.as_mut_ptr(), timeout)) };
+
+    // If we timed out, return None.
+    match result {
+        Ok(_) => Ok(Some(Event(unsafe { event.assume_init() }))),
+        Err(e) if e == crate::io::Errno::TIME => Ok(None),
```

Comment:
Is it rustix's job to handle `TIME` here, or should it just let that return to the user to handle?

Reply:
You're right, I've fixed it